src/main/java/org/broadinstitute/hellbender/tools/haplotypecaller/SimpleGenotypePriorCalculator.java

```diff
@@ -0,0 +1,131 @@
+package org.broadinstitute.hellbender.tools.haplotypecaller;
+
+import htsjdk.variant.variantcontext.Allele;
+import org.broadinstitute.hellbender.tools.walkers.genotyper.GenotypeAlleleCounts;
+import org.broadinstitute.hellbender.tools.walkers.genotyper.GenotypeCalculationArgumentCollection;
+import org.broadinstitute.hellbender.tools.walkers.genotyper.GenotypeLikelihoodCalculator;
+import org.broadinstitute.hellbender.utils.pairhmm.DragstrParams;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class SimpleGenotypePriorCalculator implements GenotypePriorCalculator {
+
+
+
+    private enum AlleleType {
+        REF, SNP, INDEL, OTHER;
+    }
+
+    private final double[] hetValues;
+    private final double[] homValues;
+    private final double[] diffValues;
+
+    private SimpleGenotypePriorCalculator(final double snpHet, final double snpHom,
+                                          final double indelHet, final double indelHom,
+                                          final double otherHet, final double otherHom) {
+        hetValues = new double[4];
```

Comment:
double[] is supposed to be better than List<Double> although in practice probrably does not make a difference (this would not be the bottle neck in HC, I guess).